lib/active_record/connection_adapters/sqlserver/schema_creation.rb

```diff
@@ -34,17 +34,20 @@ def visit_TableDefinition(o)
         end
 
         def visit_CreateIndexDefinition(o)
-          if_not_exists = o.if_not_exists
-
-          o.if_not_exists = false
-
-          sql = super
+          index = o.index
 
-          if if_not_exists
-            sql = "IF NOT EXISTS (SELECT name FROM sysindexes WHERE name = '#{o.index.name}') #{sql}"
-          end
+          sql = []
+          sql << "IF NOT EXISTS (SELECT name FROM sysindexes WHERE name = '#{o.index.name}') " if o.if_not_exists
+          sql << "CREATE"
+          sql << "UNIQUE" if index.unique
+          sql << index.type if index.type
+          sql << "INDEX"
+          sql << o.algorithm if o.algorithm
+          sql << "#{quote_column_name(index.name)} ON #{quote_table_name(index.table)}"
+          sql << "(#{quoted_columns(index)})"
+          sql << "WHERE #{index.where}" if supports_partial_index? && index.where
```

Comment:
Do we need to check for `supports_partial_index?` here? We know SQL Server does support that already.

Reply:
Fixed in PR to `main` in https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/pull/1002